### PR TITLE
[4145] Reinstate semantic logger in development and add amazing_print…

### PR DESCRIPTION
**Pull request by avinhurry**
## Context

Giving Apply the same 'beautiful logs' treatment as Find (in [this PR](https://github.com/DFE-Digital/find-teacher-training/pull/1061)) by reinstating `rails_semantic_logger` in development and adding the `amazing_print` gem.

## Changes proposed

This PR also reverts the work done in #4674.

## Screenshot

image

## Guidance to review

- Run the app locally and marvel at the logs.
- Ensure nothing has gone missing from the logs.

### CODE REVIEW

**Comment by avinhurry** [comment]
[Re-instate Semantic Logger on Apply dev and install 'Awesome print'](https://trello.com/c/uTojWv0y/4145-re-instate-semantic-logger-on-apply-dev-and-install-awesome-print)

**Review by TobyRet — COMMENTED** [review]
Logs are indeed much prettier @avinhurry  :-)
I think it would be worth @duncanjbrown talking a quick look to confirm if the semantic logger  config is correct / suitable for dev environment (as he had originally removed semantic logger from dev)

**Comment by avinhurry** [comment]
@duncanjbrown if you have some time today, can you give this PR a quick 👀 so we can get it merged in please (since you originally removed semantic logger from dev)

File: Dockerfile [hunk 1]
```diff
@@ -28,7 +28,7 @@ RUN gem update --system && \
     bundler -v && \
     bundle config set no-cache 'true' && \
     bundle config set no-binstubs 'true' && \
-    bundle --retry=5 --jobs=4 --without=development --with=production && \
+    bundle --retry=5 --jobs=4 --without=development && \
```
Comment (duncanjbrown) [inline Dockerfile:31]: Do we need to remove this?

Comment (duncanjbrown) [reply Dockerfile:31]: ah i see!

**Review by duncanjbrown — APPROVED** [review]

**Pull request merged**